### ISSUE TITLE
Support for MSSQL Managed Instance - Transparent data encryption

### ISSUE THREAD

**Issue by Marcus-James-Adams** [hashicorp/terraform-provider-azurerm#14567]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The azure portal allows you to apply TDE (Transparent data encryption) with either a system manged key, or a customer manged key,.
It would be good if we could set this from the start via terraform

I would see it working similar to mssql_server_transparent_data_encryption

### New or Affected Resource(s)

* azurerm_mssql_managed_instance_transparent_data_encryption

### Potential Terraform Configuration

Example Usage with Service Managed Key

```hcl
resource "azurerm_mssql_managed_instance_transparent_data_encryption" "example" {
  server_id = mssql_managed_instance.example.id
}
```
Example Usage with Customer Managed Key
```hcl
resource "azurerm_mssql_managed_instance_transparent_data_encryption" "example" {
  server_id = sql_managed_instance.example.id
  key_vault_key_id = azurerm_key_vault_key.example.id
  auto_rotate_key = false
}
```

* #0000

**Comment by opslivia** [comment]
This was released with [#13748](https://github.com/hashicorp/terraform-provider-azurerm/pull/13748) behind the 3.0 Beta flag.
Opt-in instructions for the 3.0 Beta [here](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-beta).

**Comment by opslivia** [comment]
Hi apologies, I linked the wrong issue closing this before. 
Since MSSQL resources have replaced the old SQL resources, I suggest we update this issue to ask for TDE on MSSQL Managed Instance instead.
